apps/emqx/src/emqx_schema_hooks.erl

```diff
@@ -7,9 +7,9 @@
 -type hookpoint() :: atom().
 
 -callback injected_fields() ->
-    #{hookpoint() => [hocon_schema:field()]}.
+    #{hookpoint() => [hocon_schema:field() | binary()]}.
```

Comment:
This change sounds unclear without an explanation (e.g. a comment): what does "injecting a string into the schema" can mean, when the function is clearly for "injecting fields"? Perhaps there's some clearer way to express it?

Reply:
Right, I see now, it's not actually about _schema_ per se. In this case it's basically a whitelist, so it sounds slightly misleading to carry it through `injected_fields`, as is renaming it to `injected_values` to be honest. I guess to make it make more sense here, we'd need something like a whole _separate schema_ for namespaced roots, consisting of a small subset of roots of the global schema. Due to complexity, this does not sound like a good idea in the context of this PR, and I don't have a better one. Maybe @savonarola might have some?